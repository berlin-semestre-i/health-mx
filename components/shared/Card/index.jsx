import React, { PureComponent } from 'react'
import { Card, Icon } from 'semantic-ui-react'
import styled from 'styled-components'
import media from 'styled-media-query'

class CustomCard extends PureComponent {
  render() {
    const { header, children, className, iconName, label, iconColor } = this.props

    return (
      <FullWidthCard className={className}>
        <Card.Content>
          <Card.Header>
            { header }
            <Secondary>
              {iconName && iconColor && (
                <Icon color={iconColor} name={iconName} />
              )}
              {iconName && !iconColor && (
                <Icon name={iconName} />
              )}
              <span> { label }</span>
            </Secondary>
          </Card.Header>
          { children }
        </Card.Content>
      </FullWidthCard>
    )
  }
}

export default CustomCard

const FullWidthCard = styled(Card)`
  && {
    width: 100%;
  }

  &&.loaderCard {
    min-height: 200px;
  }

  ${media.lessThan('medium')`
    &&.nth-col-first {
      margin-top: 1em;
    }
  `}

  &&.loaderCard .ui.loader:after {
    border-color: rgba(0,0,0,0.2) transparent transparent;
  }
`
const Secondary = styled.div`
  float: right;
  cursor: pointer;
  color: #3d5170;
  text-transform: none;
  letter-spacing: 0.5px;

  ${media.lessThan('small')`
    >span {
      display: none;
    }
  `}
`

