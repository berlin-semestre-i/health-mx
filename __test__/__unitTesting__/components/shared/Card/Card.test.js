import React from 'react'
import { shallow } from 'enzyme'
import Card from '../../../../../components/shared/Card/index'

describe('Card', () => {
  it('renders', () => {
    const component = shallow(<Card header="" />)
    expect(component).toMatchSnapshot()
  })
})