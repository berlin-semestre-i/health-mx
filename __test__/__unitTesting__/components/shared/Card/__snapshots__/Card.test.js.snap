// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders 1`] = `
<Card__FullWidthCard>
  <CardContent>
    <CardHeader />
  </CardContent>
</Card__FullWidthCard>
`;
